src/pages/settings/PasswordPage.js

```diff
@@ -59,6 +60,31 @@ class PasswordPage extends Component {
         Onyx.merge(ONYXKEYS.ACCOUNT, {error: '', success: ''});
     }
 
+    onBlurNewPassword() {
+        if (!this.state.newPassword || !this.isValidPassword()) {
+            this.setState({isPasswordRequirementsVisible: true});
+        } else {
+            this.setState({isPasswordRequirementsVisible: false});
+        }
+
+        if (this.state.newPassword && this.state.confirmNewPassword && !this.doPasswordsMatch()) {
+            this.setState({shouldShowPasswordConfirmError: true});
+        }
```

Comment:
Can this be reworked into a single if / elseif / else ladder? As is when we have a new, valid password, that does not match the confirmation password, we will call setState twice and trigger 2 rerenders.

Reply:
These both are independent checks so elseif ladder might not work, but I've updated the code [here](https://github.com/Expensify/App/pull/4409/commits/0fe950dad32f37964db928384210d28ea025c79a) to avoid rerender.